### PR TITLE
Test Modal

### PR DESCRIPTION
**Pull request by dr15**
- Adds test for the `Modal` component.
- Updates The component to make sure there is always an element to attach to.

### CODE REVIEW

File: src/components/Layout/Modal/Modal.test.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    expect(onCloseCallback).not.toHaveBeenCalled();
+  });
+
+  it('calls onClose when pressing esc', async () => {
+    const onCloseCallback = jest.fn();
+    render(renderModal({ isOpen: true, onClose: onCloseCallback }));
+
+    const modalRoot = getById(global.document.body, 'modal-root');
+    if (modalRoot) {
+      await userEvent.type(modalRoot, '{esc}');
```
Comment (dvira) [inline src/components/Layout/Modal/Modal.test.tsx:58]: You don't need the `await` or the `async` in this or any other test.

File: src/components/Layout/Modal/Modal.test.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    <div data-testid="modal-content">modal content</div>
+  </Modal>
+);
+
+describe('Modal', () => {
+  it('renders Modal content and closes modal', async () => {
+    // render modal
+    const { rerender } = render(renderModal({ isOpen: true }));
+    const modal = screen.getByTestId('modal-content');
+    expect(modal).toHaveTextContent('modal content');
```
Comment (dvira) [inline src/components/Layout/Modal/Modal.test.tsx:22]: While this is correct, it is considered best practice to use `getByText` and then explicitly assert that it is rendered, i.e.:
`const modal = screen.getByText('modal-content');
expect(modal).toBeInTheDocument();`.

File: src/components/Layout/Modal/Modal.test.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    // close modal
+    rerender(renderModal({ isOpen: false }));
+    expect(screen.queryByTestId('modal-content')).not.toBeInTheDocument();
+  });
+
+  it('calls onClose when clicking outside', async () => {
+    const onCloseCallback = jest.fn();
```
Comment (dvira) [inline src/components/Layout/Modal/Modal.test.tsx:30]: I suggest shorting this to 'onClose' just for better readability.
Please do NOT move this mock outside though, as you're using it in different tests and by moving it you would lose its contained reference (meaning you would call this **at least** once).

File: src/components/Layout/Modal/Modal.test.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { render, screen, queryByAttribute } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import Modal, { IModal } from '.';
+
+type ITestModalProps = Omit<IModal, 'children'>;
```
Comment (dvira) [inline src/components/Layout/Modal/Modal.test.tsx:7]: Not that important, but still was wondering, didn't we agree on some convention regarding naming props? I think it was prefix with `I` and no suffix.

**Review by dvira — CHANGES_REQUESTED** [review]
Looks good! I had a few minor comments.
Please provide a screenshot of the updated coverage if you can.

File: src/components/Layout/Modal/Modal.test.tsx (hunk 4 above)
Comment (dr15) [reply src/components/Layout/Modal/Modal.test.tsx:7]: As far as I know it's both

File: src/components/Layout/Modal/Modal.test.tsx (hunk 1 above)
Comment (dr15) [reply src/components/Layout/Modal/Modal.test.tsx:58]: Fixed

File: src/components/Layout/Modal/Modal.test.tsx (hunk 2 above)
Comment (dr15) [reply src/components/Layout/Modal/Modal.test.tsx:22]: Fixed

File: src/components/Layout/Modal/Modal.test.tsx (hunk 3 above)
Comment (dr15) [reply src/components/Layout/Modal/Modal.test.tsx:50]: Ok

**Review by dvira — APPROVED** [review]

**Pull request merged**